ambry-replication/src/main/java/com/github/ambry/replication/BackupCheckerThread.java

```diff
@@ -59,6 +72,22 @@ public class BackupCheckerThread extends ReplicaThread {
 
   private final Logger logger = LoggerFactory.getLogger(BackupCheckerThread.class);
   public static final String DR_Verifier_Keyword = "dr";
+  protected final AmbryCache fileDescriptorCache;
+  private final String basePath = "/export/content/lid/logs/ambry-server/i001";
+  public static final DateFormat dateFormat = new SimpleDateFormat("dd MMM yyyy HH:mm:ss:SSS");
+
+  protected class FileDescriptor implements AmbryCacheEntry {
```

Comment:
Why does a `FileDescriptor` class need to implement `AmbryCacheEntry` ?

Reply:
If this is not done, the code compiles but fails during runtime with an error. The AmbryCache expects the objects inserted to be of type AmbryCacheEntry.

Reply:
I assume we only open one or a couple files to save the outcome. File descriptor limit shouldn't be our concern, right?

Reply:
not one or two. For the osua event, there are 40 partitions. For each partition, there are 3 replicas. And each replica will have two files - missingKeys and replicaCheckStatus. So 40*3*2 = 240 files at most. And this can grow if the number of partitions increase. I also plan to add one more file later.